internal/ingress/controller/controller.go

```diff
@@ -1132,6 +1134,11 @@ func (n *NGINXController) createServers(data []*ingress.Ingress,
 		}
 	}
 
+	if n.cfg.DisableCatchAll {
```

Comment:
Would it be better to stop the creation of the default server here https://github.com/Shopify/ingress/blob/75788e1c1113728bf48abbe58e7a1a985f14915b/internal/ingress/controller/controller.go#L929-L943 rather than delete it after creation? I guess the former could be more prone to bugs but would be more explicit when future changes are made.

Reply:
I'll have a look 👍

Reply:
Here is what it looks like, wdyt? https://github.com/Shopify/ingress/compare/master...Shopify:disable-catch-all-2?expand=1